tools/flaky-test-retryer/handler.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// handler.go contains most of the main logic for the flaky-test-retryer. Listen for
+// incoming Pubsub messages, verify that the message we received is one we want to
+// process, compare flaky and failed tests, and trigger retests if necessary.
+
+package main
+
+import (
+	"context"
+	"log"
+
+	"github.com/knative/test-infra/tools/monitoring/subscriber"
+	// TODO: remove this import once "k8s.io/test-infra" import problems are fixed
+	// https://github.com/test-infra/test-infra/issues/912
+	"github.com/knative/test-infra/tools/monitoring/prowapi"
+)
+
+// HandlerClient wraps the other clients we need when processing failed jobs.
+type HandlerClient struct {
+	context.Context
+	pubsub *subscriber.Client
+	github *GithubClient
+}
+
+func NewHandlerClient(githubClient *GithubClient) (*HandlerClient, error) {
```

Comment:
any reason to pass this in? Why cant we create it here instead of main and passing it in?

Reply:
fixed